test/external/guardian/liquidationWithdrawalKeyHijacking.ts

```diff
@@ -0,0 +1,479 @@
+      expect(liquidAccountTwoBeforeBalances.get(core.marketIds.weth.toString())?.isZero()).to.be.false;
+      expect(liquidAccountTwoBeforeBalances.get(core.marketIds.nativeUsdc!.toString()) === undefined).to.be.true; // balance 0 but javavascript
+
+      console.log(`Liquidating account ${liquidAccount.number} but using accounts' ${borrowAccountNumber2} key: ${secondBorrowerKey}`);
+      await executeProxyLiquidation(
+        solidAccount, 
+        liquidAccount, 
+        [secondBorrowerKey], 
+        [], 
+        _amountWeiForLiquidationOne, 
+        _wethAmountOne
+      );
+
+      console.log("Balances for accounts after liquidations show no change to second account and first account has only liquidation output token");
+      const liquidAccountOneAfterBalances = await getMarketBalances(liquidAccount);
+      expect(liquidAccountOneAfterBalances.get(marketId.toString()) === undefined).to.be.true; // balance 0 but javavascript
+      expect(liquidAccountOneAfterBalances.get(core.marketIds.weth.toString()) === undefined).to.be.true; // balance 0 but javavascript
+      expect(liquidAccountOneAfterBalances.get(core.marketIds.nativeUsdc!.toString())?.isZero()).to.be.false; // liquidation output token was in USDC
+
+      const liquidAccountTwoAfterBalances = await getMarketBalances(liquidAccount2);
+      expect(liquidAccountTwoAfterBalances.get(marketId.toString())).to.be.equal(liquidAccountTwoBeforeBalances.get(marketId.toString()));
+      expect(liquidAccountTwoAfterBalances.get(core.marketIds.weth.toString())).to.be.equal(liquidAccountTwoBeforeBalances.get(core.marketIds.weth.toString()));
+      expect(liquidAccountTwoAfterBalances.get(core.marketIds.nativeUsdc!.toString())).to.be.equal(liquidAccountTwoBeforeBalances.get(core.marketIds.nativeUsdc!.toString()));
+
+      console.log(`Since account ${liquidAccount.number} is now over-collateralized, the operation to retrieve his failed withdrawal can initiated by a trusted handler`);      
+      
+      await gmxV2Registry.connect(core.governance).ownerSetIsHandler(core.hhUser1.address, true);
+      const unwrapperAsTrustedHandler = unwrapper.connect(core.hhUser1);
+      const result = await unwrapperAsTrustedHandler.executeWithdrawalForRetry(firstBorrowerKey, {gasLimit: 30_000_000});
+      await result.wait();
+      console.log("But this operation will fail since adding the extra funds from the initial key would be interpreted by DolomiteMargin as an increase in borrowing from the GM market, which is a closed market");
+      await expectEvent(eventEmitter, result, 'AsyncWithdrawalFailed', {
+        key: firstBorrowerKey,
+        token: factory.address,
+        reason: `OperationImpl: Market is closing <${marketId}>`
+      });
+
+      console.log("The second position has also become over-collateralize during the price variation so continuing liquidation will fail regardless of withdrawal key");
```

Comment:
nit: over-collateralized

By "price variation" you are referring to ` const decreasedWethPrice = initialWethPrice.mul(35).div(100);`?